typing/includemod.ml

```diff
@@ -79,7 +77,10 @@ module Error = struct
   and arg_functor_param_symptom =
     (functor_parameter, Ident.t) functor_param_symptom
 
-  and functor_params_diff = (functor_parameter list * module_type) core_diff
+  and functor_params_symptom = bool
```

Comment:
there is a comment in `includemod.mli`.

(The latest commit no longer uses boolean anyway)

Reply:
As discussed, this PR cares about the presense/absense of modes, not module vs. module type, and because functor parameters carries their own modes, we don't need to record this.